awx/main/tests/functional/test_inventory_source_injectors.py

```diff
@@ -192,10 +193,16 @@ def create_reference_data(source_dir, env, content):
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('this_kind', CLOUD_PROVIDERS)
-def test_inventory_update_injected_content(this_kind, inventory, fake_credential_factory, mock_me):
+@pytest('this_kind', discover_available_cloud_provider_plugin_names)
+def test_inventory_update_injected_content(this_kind, inventory, fake_credential_factory, self):
     ExecutionEnvironment.objects.create(name='Control Plane EE', managed=True)
     ExecutionEnvironment.objects.create(name='Default Job EE', managed=False)
+    cloud_providers = discover_available_cloud_provider_plugin_names()
+    for this_kind in cloud_providers:
+        with self.subTest(this_kind=this_kind):
+            injector = InventorySource.injectors[this_kind]
+            if injector.plugin_name is None:
+                self.skipTest(f'Use of inventory plugin is not enabled for source {this_kind}')
```

Comment:
Same here, it's a `unittest` helper, not something that would work in `pytest. Plus, `self` means needing a class instance. Besides, if skipping succeeded, it'd interrupt the entire test function, not just a part of it. If you need to filter things out, do this in the preparation stage, only keeping things you need in the iterable.